main/settings.py

```diff
@@ -872,3 +876,77 @@ def get_all_config_keys():
     "PASSWORD_RESET_SHOW_EMAIL_NOT_FOUND": True,
     "EMAIL": {"password_reset": "authentication.views.CustomPasswordResetEmail"},
 }
+
+# NovoEd SAML settings (using Bootcamps app as IdP)
+IDP_BASE_URL = urljoin(BOOTCAMP_ECOMMERCE_BASE_URL, "/idp")
+_novoed_saml_key = get_string(
+    "NOVOED_SAML_KEY",
+    None,
+    description="Contents of the SAML key for NovoEd ('\n' line separators)",
+)
+_novoed_saml_cert = get_string(
+    "NOVOED_SAML_CERT",
+    None,
+    description="Contents of the SAML certificate for NovoEd ('\n' line separators)",
+)
+NOVOED_SAML_DEBUG = get_bool(
+    "NOVOED_SAML_DEBUG",
+    False,
+    description="Flag indicating whether the SAML config should be set to debug mode",
+)
+NOVOED_SAML_CONFIG_TTL_HOURS = get_int(
+    "NOVOED_SAML_CONFIG_TTL_HOURS",
+    365 * 24,
+    description="The number of hours that the SAML config is expected to be accurate",
+)
+
+if _novoed_saml_key and _novoed_saml_cert:
+    with NamedTemporaryFile(prefix="saml_", suffix=".key", delete=False) as f:
+        lines = _novoed_saml_key.split("\\n")
+        for line in lines:
+            f.write(f"{line}\n".encode("utf-8"))
+        _novoed_saml_key_file_path = f.name
+    with NamedTemporaryFile(prefix="saml_", suffix=".cert", delete=False) as f:
+        lines = _novoed_saml_cert.split("\\n")
+        for line in lines:
+            f.write(f"{line}\n".encode("utf-8"))
+        _novoed_saml_cert_file_path = f.name
+    SAML_IDP_CONFIG = {
+        "debug": NOVOED_SAML_DEBUG,
+        "xmlsec_binary": get_xmlsec_binary(
+            ["/usr/bin/xmlsec1", "/usr/local/bin/xmlsec1"]
+        ),
+        "entityid": f"{IDP_BASE_URL}/metadata",
+        "description": "MIT Bootcamps",
+        "service": {
+            "idp": {
+                "name": "Bootcamp IdP",
+                "endpoints": {
+                    "single_sign_on_service": [
+                        (f"{IDP_BASE_URL}/sso/post", saml2.BINDING_HTTP_POST),
+                        (f"{IDP_BASE_URL}/sso/redirect", saml2.BINDING_HTTP_REDIRECT),
+                    ],
+                    "single_logout_service": [
+                        (f"{IDP_BASE_URL}/slo/post", saml2.BINDING_HTTP_POST),
+                        (f"{IDP_BASE_URL}/slo/redirect", saml2.BINDING_HTTP_REDIRECT),
+                    ],
+                },
+                "name_id_format": [NAMEID_FORMAT_UNSPECIFIED, NAMEID_FORMAT_PERSISTENT],
+                "sign_response": False,
```

Comment:
Does this need to be false? In the example code in the docs it's set to true, but maybe the requirements are different here

Reply:
This setting might not actually be needed since this is an IdP config, not an SP. I'll test

Reply:
@noisecapella Actually this is just a default. The `ServiceProvider` record has a property that overrides this if you specify it (and we do). No risk leaving this as `False` for the moment